R/bs4_book.R

```diff
@@ -390,16 +415,35 @@ tweak_tables <- function(html) {
 
 tweak_navbar <- function(html, toc, active = "", rmd_index = NULL, repo = NULL) {
 
+  if (!is.null(repo) && length(repo) == 1) {
+    repo <- list(
+      base = repo,
+      branch = "master",
+      subdir = NULL
+    )
+  }
+
   # Source links ------------------------------------------------------------
   if (!is.null(repo) && active %in% names(rmd_index)) {
-    repo_edit <- paste0(repo, "/edit/master/", rmd_index[[active]])
-    repo_view <- paste0(repo, "/blob/master/", rmd_index[[active]])
+
+    if (!is.null(repo$subdir)) {
+      repo$subdir <- paste0(repo$subdir, "/")
+    }
+
+    repo_edit <- paste0(repo$base, "/edit/", repo$branch,"/", repo$subdir, rmd_index[[active]])
+    repo_view <- paste0(repo$base, "/blob/", repo$branch,"/", repo$subdir, rmd_index[[active]])
   } else {
     repo_edit <- NULL
     repo_view <- NULL
   }
 
-  template_link(html, ".//a[@id='book-repo']", repo)
+  if (!is.null(repo$base) && !grepl("github\\.com", repo$base)) {
+    template_link_icon(html, ".//a[@id='book-repo']", "fab fa-gitlab")
+    template_link_icon(html, ".//a[@id='book-source']", "fab fa-gitlab")
+    template_link_icon(html, ".//a[@id='book-edit']", "fab fa-gitlab")
+  }
```

Comment:
Why don't we offer a parameter to choose the fa icon to use ? 
instead of just assuming non github = gitlab ?